### PR TITLE
Create release PR action

### PR DESCRIPTION
**Pull request by StephenHulme**
Automatically create and update a release to master PR, allowing for simpler release review and creation.

#### Changes proposed in this pull request

- Add a workflow to create and update a merge-to-master PR each time a PR is merged into develop

Known Issues:
- Downstream `pull_request` events are not triggered by a pull request ♻️ 
  - This is currently preventing the `Check release version` action from being run in the resulting PR, only when there is a single commit in the diff. Subsequent pushes to `develop` will trigger the `pull_request` event.
  - Edit: this known issue is fixed in #1970 

See the created PRs: https://github.com/sanger/limber/pull/1940, https://github.com/sanger/limber/pull/1957

#### Instructions for Reviewers

_[All PRs] - Confirm PR template filled_  
_[Feature Branches] - Review code_  
_[Production Merges to `main`]_  
 &nbsp; &nbsp; \- _Check story numbers included_  
 &nbsp; &nbsp; \- _Check for debug code_  
 &nbsp; &nbsp; \- _Check version_

### CODE REVIEW

File: .github/workflows/create_release_pr.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+          &nbsp; &nbsp; \- _Check version is appropiate_
+
+          This description will be overwritten on the next push
+          This PR was auto-generated on $(TZ=Europe/London date --iso-8601=seconds)"
+
+          # Check if a PR already exists
+          EXISTING_PR=$(gh pr list --base master --head develop --json number --jq '.[0].number')
+
+          if [ -z "$EXISTING_PR" ]; then
+            gh pr create --base master --head develop --title "Merge Develop into Master" --body "$PR_BODY"
```
Comment (harrietc52) [inline .github/workflows/create_release_pr.yml:37]: Maybe the PR title could include “Automated” just to make it clear that this one was auto creaetd

File: .github/workflows/create_release_pr.yml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+          PR_BODY="#### Changes proposed in this pull request
+
+          $(echo "$STORIES")
+
+          #### Instructions for Reviewers
+
+          &nbsp; &nbsp; \- _Check stories are ready for release_
+          &nbsp; &nbsp; \- _Check for any database migrations_
+          &nbsp; &nbsp; \- _Check for debug code_
+          &nbsp; &nbsp; \- _Check version is appropiate_
```
Comment (sdjmchattie) [inline .github/workflows/create_release_pr.yml:50]:
```suggestion
          - _Check stories are ready for release._
          - _Check for any database migrations._
          - _Check for debug code._
          - _Check version is appropriate._
```

**Review by sdjmchattie — APPROVED** [review]
One suggestion to remove unnecessary HTML from what is basically a bulleted list.

**Review by dasunpubudumal — APPROVED** [review]
I did have a look at this but just realised that I haven't approved it. Sorry!

It's good to have a manual dispatching workflow to create the master PR. Thanks!

**Pull request merged**